src/pkg/task/hook.go

```diff
@@ -37,31 +45,63 @@ type HookHandler struct {
 }
 
 // Handle the job status changing webhook
-func (h *HookHandler) Handle(ctx context.Context, taskID int64, sc *job.StatusChange) error {
-	task, err := h.taskDAO.Get(ctx, taskID)
+func (h *HookHandler) Handle(ctx context.Context, sc *job.StatusChange) error {
+	logger := log.GetLogger(ctx)
+
+	jobID := sc.JobID
+	if sc.Metadata.JobKind == job.KindScheduled {
+		jobID = sc.Metadata.UpstreamJobID
+	}
+	tasks, err := h.taskDAO.List(ctx, &q.Query{
+		Keywords: map[string]interface{}{
+			"JobID": jobID,
+		},
+	})
+	if err != nil {
+		return err
+	}
+	if len(tasks) == 0 {
+		return errors.New(nil).WithCode(errors.NotFoundCode).
```

Comment:
If this error can not be fixed, maybe we just log the error and return 200 to avoid js retrying

Reply:
The error is ignored by the API handler: `src/server/handler/job_status_hook.go`

Reply:
ok